WeatherForecast/WeatherForecast/Feature/Weather/Model/Repository/WeatherRepository.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  WeatherRepository.swift
+//  WeatherForecast
+//
+//  Created by Choi Oliver on 3/13/24.
+//
+
+import UIKit
+import Combine
+
+protocol WeatherRepository {
+    func fetchWeather() -> AnyPublisher<FetchWeatherResult, Error>
+}
+
+struct WeatherRepositoryImpl: WeatherRepository {
+    func fetchWeather() -> AnyPublisher<FetchWeatherResult, Error> {
```

Comment:
```suggestion
  protocol WeatherDataSource {
    func fetchWeatherData() -> AnyPublisher<Data, Error>
}

class WeatherFetcher {
    private let dataSource: WeatherDataSource

    init(dataSource: WeatherDataSource) {
        self.dataSource = dataSource
    }

    func fetchWeather() -> AnyPublisher<FetchWeatherResult, Error> {
        dataSource.fetchWeatherData()
            .decode(type: FetchWeatherResultDTO.self, decoder: JSONDecoder())
            .map { FetchWeatherResult(dto: $0) }
            .eraseToAnyPublisher()
    }
}

```

먼저, 저도 컴바인은 익숙하지 않은 부분이 있습니다.
이런식으로 구현을 생각해보았는데요.
이렇게 했을때 혹여나 문제가 될 부분이 있는것 같으시면 언제든 의견주세요 !

(WeatherDataSource는 현재 수준에서 오버엔지니어링일수 있을것 같긴하네요.
테스트까지 고려해본다면 WeatherDataSource 프로토콜을 사용함으로써, mock또는 stub을 구현하여, 
실제 데이터 소스 대신 테스트용 데이터를 반환하게 할 수 있을것 같아요.)